cmd/controller-manager/app/options/options.go

```diff
@@ -153,40 +171,46 @@ func NewDefaultControllerManagerComponentConfig(insecurePort int32) componentcon
 
 // AddFlags adds common/default flags for both the kube and cloud Controller Manager Server to the
 // specified FlagSet. Any common changes should be made here. Any individual changes should be made in that controller.
-func (o *GenericControllerManagerOptions) AddFlags(fs *pflag.FlagSet) {
-	fs.BoolVar(&o.ComponentConfig.UseServiceAccountCredentials, "use-service-account-credentials", o.ComponentConfig.UseServiceAccountCredentials, "If true, use individual service account credentials for each controller.")
-	fs.StringVar(&o.ComponentConfig.CloudConfigFile, "cloud-config", o.ComponentConfig.CloudConfigFile, "The path to the cloud provider configuration file. Empty string for no configuration file.")
-	fs.BoolVar(&o.ComponentConfig.AllowUntaggedCloud, "allow-untagged-cloud", false, "Allow the cluster to run without the cluster-id on cloud instances. This is a legacy mode of operation and a cluster-id will be required in the future.")
+func (o *ControllerManagerOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.Int32Var(&o.ConcurrentServiceSyncs, "concurrent-service-syncs", o.ConcurrentServiceSyncs, "The number of services that are allowed to sync concurrently. Larger number = more responsive service management, but more CPU (and network) load")
+	// TODO: remove --service-account-private-key-file 6 months after 1.8 is released (~1.10)
+	fs.StringVar(&o.ServiceAccountKeyFile, "service-account-private-key-file", o.ServiceAccountKeyFile, "Filename containing a PEM-encoded private RSA or ECDSA key used to sign service account tokens.")
```

Comment:
hadn't these been specific to the cloud controller-manager? Why are they now in the generic struct?

Reply:
Those two flag has exits in kube-controller-manager, so i remove those. sorry, i will revert it